major: Cleanup

JamMassey opened the pull request:
Going through debugging/testing now, but just so you can see diff or if you wanted to jump in

Commit:
trying to setup so he runs both loacally and on dev, also included a bump and semantic release pipeline - working to the long term goal of making him super super easily runabble, configuarable and deployable. Currently in a broken state, trying to remove the _settings.py file we have under cogs as this means that the settings can ONLY be instantiated with an environment variable, id like to have the ability to make this a little easier for local devs

Commit:
update setup.cfg

Commit:
way way closer to working. trying to feed the server_setting toml through to the cogs by applying it as an attribute on the bot during the main function. this works much like how the db is integrated

Commit:
changed env variable name back to old

Commit:
versioneer files

Commit:
merge with dev

Commit:
merge with dev

Commit:
now runs locally and has a flag to drop the database on launch so you can run the bot completely w/o docker, ive also added a semantic release pipeline that triggers based off of pr titles, refactored the server settings tomls such that theyre fed in from top level to meet a design pattern closer to dependency injection. can now be launched with 'pip install .' and then simply hitting 'zorak'.

Commit:
update comment

Commit:
update comment

Commit:
updated readme

Commit:
seen xarlos added new dependencies, so updated to match

Xarlos89 reviewed (changes requested):
very cool changes, unnecessary extra template and we gotta sort the issue with the 'zorak' call not working in the docker set up

Commit:
made all review changes

JamMassey committed:
Delete debug.py

JamMassey committed:
Delete .pre-commit-config.yaml

Xarlos89 committed:
Update README.md

Updates and cleanup now that things are changing. Must keep the Readme up to date.
D O C U M E N T A T I O N

Xarlos89 committed:
Merge issue in a few files.
